pkg/virt-controller/services/template.go

```diff
@@ -533,8 +533,8 @@ func (t *templateService) RenderLaunchManifest(vmi *v1.VirtualMachineInstance) (
 				})
 				volumeMounts = append(volumeMounts, k8sv1.VolumeMount{
 					Name:      volumeName,
-					MountPath: filepath.Join(config.SecretSourceDir, volume.Name, "userdata"),
-					SubPath:   "userdata",
+					MountPath: filepath.Join(config.SecretSourceDir, volume.Name, "userData"),
+					SubPath:   "userData",
```

Comment:
doesn't this break backwards compatibility? From what I gather, this change would mean existing cloud-init secrets that are in production wouldn't work, right?

Reply:
Right! I created this PR to check if any functional test would catch that.
However I then decided to find and run the right test(s) locally instead of using hours of test lane cpu time.
I was hopeful when I found an upgrade test using this: https://github.com/kubevirt/kubevirt/blob/master/tests/operator_test.go#L551
but unfortunately it passes.
I want to first find/write a test that breaks with this change, then handle backwards compatibility (probably by checking for both spellings)

Reply:
i'd think the tests in `tests/vmi_cloudinit_test.go` that set the 'userdata' key in a secret would fail. That might be a good place to look.

besides that. I'm curious why we should worry about making this change and adding the additional complexity involved with backwards compatibility? If we're worried about adhering to camel case, the `networkdata` secret key for cloud-init has the same issue as `userdata`.

Is there any more context around why you think this needs to change?

Reply:
Thanks for the pointer.
I did notice networkdata too, and was planning on adding it after userdata is dealt with.
The context for this change is compatibility with a downstream project that handles secrets creation before starting VMs.
This could obviously be fixed downstream instead, but it seems to me that camel case is more appropriate here.